File: Practice/20/Python/twentiethTask.py
class Alcohol:
    def __init__(self):
        self.value = 0.0
        self.volume = 0.0
        self.name = ""


money = int(input())
n = int(input())

result = Alcohol()
inp = Alcohol()

for i in range(n):
    inp.name = input()
    inp.value = int(input())
    inp.volume = int(input())
    liters = (money // inp.value) * inp.volume
    if liters == 0:
        continue

    if result.value == 0:
        result = inp
        continue
    result_liters = (money // result.value) * result.volume
    if liters > result_liters:
        result = inp

if result.value == 0:
    print(-1)
else:
    bottles = money // result.value
    print(result.name, bottles, "\n", bottles * result.volume, "\n",
           money - result.value * bottles)

File: Practice/20/C++/twentiethTask/twentiethTask.cpp
#include <iostream>
using namespace std;

struct Alcohol
	{
		string name;
		int value;
		double volume;
	};

int main()
{
	int money, n;
	cin >> money >> n;
	Alcohol input;
	Alcohol result;
	for (int i = 0; i < n; i++)
	{
		cin >> input.name >> input.value >> input.volume;
		int liters = (money / input.value) * input.volume;
		if (input.value > money)
		{
			cout << -1 << endl;
			return 0;
		}
		else;
		int result_liters = (money / result.value) * result.volume;
		if (liters > result_liters)
			result = input;
	}
		int bottles_count = money / result.value;
		cout << result.name << " " << bottles_count << endl;
		cout << bottles_count * result.volume << endl;
		cout << money - result.value * bottles_count << endl;
}